Cannot build solution locally

### Description

As part of #4483 I tried to build the solution locally, but package restore fails.

At first it failed with this error:

```
C:\Coding\dotnet\extensions\eng\common\internal\Tools.csproj : error NU1100: Unable to resolve 'Microsoft.ManifestTool.CrossPlatform (>= 2.1.3)' for '.NETFramework,Version=v4.7.2'. PackageSourceMapping is enabled, the following source(s) were not considered: dotnet-core-internal-tooling. [C:\Coding\dotnet\extensions\extensions.sln]
C:\Coding\dotnet\extensions\eng\common\internal\Tools.csproj : error NU1100: Unable to resolve 'Microsoft.VisualStudioEng.MicroBuild.Core (>= 1.0.0)' for '.NETFramework,Version=v4.7.2'. PackageSourceMapping is enabled, the following source(s) were not considered: dotnet-core-internal-tooling. [C:\Coding\dotnet\extensions\extensions.sln]
C:\Coding\dotnet\extensions\eng\common\internal\Tools.csproj : error NU1100: Unable to resolve 'Microsoft.VisualStudioEng.MicroBuild.Plugins.SwixBuild (>= 1.1.286)' for '.NETFramework,Version=v4.7.2'. PackageSourceMapping is enabled, the following source(s) were not considered: dotnet-core-internal-tooling. [C:\Coding\dotnet\extensions\extensions.sln]
```

Then I edited NuGet.config to add the following:

```
<packageSource key="dotnet-core-internal-tooling">
  <package pattern="*" />
</packageSource>
```

Now the build fails with the following error because I don't have access to the AzDo package feed for DevDiv:

```
C:\Coding\dotnet\extensions\eng\common\internal\Tools.csproj : error NU1301: Unable to load the service index for source https://pkgs.dev.azure.com/devdiv/_packaging/dotnet-core-internal-tooling/nuget/v3/index.json. [C:\Coding\dotnet\extensions\extensions.sln]
```

### Reproduction Steps

Run `restore.cmd` while not being a Microsoft employee.

### Expected behavior

The solution successfully builds.

### Actual behavior

The solution fails to restore NuGet packages.

### Regression?

Unsure.

### Known Workarounds

Get hired by Microsoft to work in DevDiv 😄


### Configuration

8.0.100-rc.2.23462.28

### Other information

_No response_

According to the linked document in the README on how to build the solution:

https://github.com/dotnet/extensions/blob/71a221002f15ba8fcbfea55bf6adb2e63861e56d/docs/building.md?plain=1#L68-L78

This then fails as described in the issue description above:

```
C:\Coding\dotnet\extensions\eng\common\internal\Tools.csproj : error NU1100: Unable to resolve 'Microsoft.Manifest
Tool.CrossPlatform (>= 2.1.3)' for '.NETFramework,Version=v4.7.2'. PackageSourceMapping is enabled, the following
source(s) were not considered: dotnet-core-internal-tooling. [C:\Coding\dotnet\extensions\extensions.sln]
C:\Coding\dotnet\extensions\eng\common\internal\Tools.csproj : error NU1100: Unable to resolve 'Microsoft.VisualSt
udioEng.MicroBuild.Core (>= 1.0.0)' for '.NETFramework,Version=v4.7.2'. PackageSourceMapping is enabled, the follo
wing source(s) were not considered: dotnet-core-internal-tooling. [C:\Coding\dotnet\extensions\extensions.sln]
C:\Coding\dotnet\extensions\eng\common\internal\Tools.csproj : error NU1100: Unable to resolve 'Microsoft.VisualSt
udioEng.MicroBuild.Plugins.SwixBuild (>= 1.1.286)' for '.NETFramework,Version=v4.7.2'. PackageSourceMapping is ena
bled, the following source(s) were not considered: dotnet-core-internal-tooling. [C:\Coding\dotnet\extensions\exte
nsions.sln]
C:\Coding\dotnet\extensions\.dotnet\sdk\9.0.100-alpha.1.23476.6\NuGet.RestoreEx.targets(19,5): error : Unable to l
oad the service index for source https://pkgs.dev.azure.com/devdiv/_packaging/dotnet-core-internal-tooling/nuget/v
3/index.json. [C:\Coding\dotnet\extensions\extensions.sln]
C:\Coding\dotnet\extensions\.dotnet\sdk\9.0.100-alpha.1.23476.6\NuGet.RestoreEx.targets(19,5): error :    at NuGet
.Protocol.ServiceIndexResourceV3Provider.GetServiceIndexResourceV3(SourceRepository source, DateTime utcNow, ILogg
er log, CancellationToken token) [C:\Coding\dotnet\extensions\extensions.sln]
C:\Coding\dotnet\extensions\.dotnet\sdk\9.0.100-alpha.1.23476.6\NuGet.RestoreEx.targets(19,5): error :    at NuGet
.Protocol.ServiceIndexResourceV3Provider.TryCreate(SourceRepository source, CancellationToken token) [C:\Coding\do
tnet\extensions\extensions.sln]
C:\Coding\dotnet\extensions\.dotnet\sdk\9.0.100-alpha.1.23476.6\NuGet.RestoreEx.targets(19,5): error :    at NuGet
.Protocol.Core.Types.SourceRepository.GetResourceAsync[T](CancellationToken token) [C:\Coding\dotnet\extensions\ex
tensions.sln]
C:\Coding\dotnet\extensions\.dotnet\sdk\9.0.100-alpha.1.23476.6\NuGet.RestoreEx.targets(19,5): error :    at NuGet
.Protocol.Providers.VulnerabilityInfoResourceV3Provider.TryCreate(SourceRepository source, CancellationToken token
) [C:\Coding\dotnet\extensions\extensions.sln]
C:\Coding\dotnet\extensions\.dotnet\sdk\9.0.100-alpha.1.23476.6\NuGet.RestoreEx.targets(19,5): error :    at NuGet
.Protocol.Core.Types.SourceRepository.GetResourceAsync[T](CancellationToken token) [C:\Coding\dotnet\extensions\ex
tensions.sln]
C:\Coding\dotnet\extensions\.dotnet\sdk\9.0.100-alpha.1.23476.6\NuGet.RestoreEx.targets(19,5): error :    at NuGet
.Commands.VulnerabilityInformationProvider.GetVulnerabilityInfoAsync() [C:\Coding\dotnet\extensions\extensions.sln
]
C:\Coding\dotnet\extensions\.dotnet\sdk\9.0.100-alpha.1.23476.6\NuGet.RestoreEx.targets(19,5): error :    at NuGet
.Commands.VulnerabilityInformationProvider.GetVulnerabilityInformationAsync(CancellationToken cancellationToken) [
C:\Coding\dotnet\extensions\extensions.sln]
C:\Coding\dotnet\extensions\.dotnet\sdk\9.0.100-alpha.1.23476.6\NuGet.RestoreEx.targets(19,5): error :    at NuGet
.Commands.Restore.Utility.AuditUtility.GetAllVulnerabilityDataAsync(CancellationToken cancellationToken) [C:\Codin
g\dotnet\extensions\extensions.sln]
C:\Coding\dotnet\extensions\.dotnet\sdk\9.0.100-alpha.1.23476.6\NuGet.RestoreEx.targets(19,5): error :    at NuGet
.Commands.Restore.Utility.AuditUtility.CheckPackageVulnerabilitiesAsync(CancellationToken cancellationToken) [C:\C
oding\dotnet\extensions\extensions.sln]
C:\Coding\dotnet\extensions\.dotnet\sdk\9.0.100-alpha.1.23476.6\NuGet.RestoreEx.targets(19,5): error :    at NuGet
.Commands.RestoreCommand.PerformAuditAsync(EnabledValue enableAudit, IEnumerable`1 graphs, TelemetryActivity telem
etry, CancellationToken token) [C:\Coding\dotnet\extensions\extensions.sln]
C:\Coding\dotnet\extensions\.dotnet\sdk\9.0.100-alpha.1.23476.6\NuGet.RestoreEx.targets(19,5): error :    at NuGet
.Commands.RestoreCommand.ExecuteAsync(CancellationToken token) [C:\Coding\dotnet\extensions\extensions.sln]
C:\Coding\dotnet\extensions\.dotnet\sdk\9.0.100-alpha.1.23476.6\NuGet.RestoreEx.targets(19,5): error :    at NuGet
.Commands.RestoreRunner.ExecuteAsync(RestoreSummaryRequest summaryRequest, CancellationToken token) [C:\Coding\dot
net\extensions\extensions.sln]
C:\Coding\dotnet\extensions\.dotnet\sdk\9.0.100-alpha.1.23476.6\NuGet.RestoreEx.targets(19,5): error :    at NuGet
.Commands.RestoreRunner.ExecuteAndCommitAsync(RestoreSummaryRequest summaryRequest, IRestoreProgressReporter progr
essReporter, CancellationToken token) [C:\Coding\dotnet\extensions\extensions.sln]
C:\Coding\dotnet\extensions\.dotnet\sdk\9.0.100-alpha.1.23476.6\NuGet.RestoreEx.targets(19,5): error :    at NuGet
.Commands.RestoreRunner.CompleteTaskAsync(List`1 restoreTasks) [C:\Coding\dotnet\extensions\extensions.sln]
C:\Coding\dotnet\extensions\.dotnet\sdk\9.0.100-alpha.1.23476.6\NuGet.RestoreEx.targets(19,5): error :    at NuGet
.Commands.RestoreRunner.RunAsync(IEnumerable`1 restoreRequests, RestoreArgs restoreArgs, CancellationToken token)
[C:\Coding\dotnet\extensions\extensions.sln]
C:\Coding\dotnet\extensions\.dotnet\sdk\9.0.100-alpha.1.23476.6\NuGet.RestoreEx.targets(19,5): error :    at NuGet
.Commands.RestoreRunner.RunAsync(RestoreArgs restoreContext, CancellationToken token) [C:\Coding\dotnet\extensions
\extensions.sln]
C:\Coding\dotnet\extensions\.dotnet\sdk\9.0.100-alpha.1.23476.6\NuGet.RestoreEx.targets(19,5): error :    at NuGet
.Build.Tasks.BuildTasksUtility.RestoreAsync(DependencyGraphSpec dependencyGraphSpec, Boolean interactive, Boolean
recursive, Boolean noCache, Boolean ignoreFailedSources, Boolean disableParallel, Boolean force, Boolean forceEval
uate, Boolean hideWarningsAndErrors, Boolean restorePC, Boolean cleanupAssetsForUnsupportedProjects, ILogger log,
CancellationToken cancellationToken) [C:\Coding\dotnet\extensions\extensions.sln]
C:\Coding\dotnet\extensions\.dotnet\sdk\9.0.100-alpha.1.23476.6\NuGet.RestoreEx.targets(19,5): error :    at NuGet
.Build.Tasks.Console.MSBuildStaticGraphRestore.RestoreAsync(String entryProjectFilePath, IDictionary`2 globalPrope
rties, IReadOnlyDictionary`2 options) [C:\Coding\dotnet\extensions\extensions.sln]
    0 Warning(s)
    4 Error(s)

Time Elapsed 00:00:27.08
Build failed with exit code 1. Check errors above.
```

--- a/.gitignore
+++ b/.gitignore
@@ -38,6 +38,7 @@ bld/
 .vs/
 .build/
 .vscode
+!.vscode/settings.json
 
 # MSTest test Results
 [Tt]est[Rr]esult*/

--- a/.vscode/settings.json
+++ b/.vscode/settings.json
@@ -1,0 +1,3 @@
+{
+  "dotnet.automaticallyCreateSolutionInWorkspace": false
+}

